src/core/net/nd6.cpp

```diff
@@ -182,6 +182,69 @@ void RaFlagsExtOption::Init(void)
     OT_UNUSED_VARIABLE(mFlags);
 }
 
+//----------------------------------------------------------------------------------------------------------------------
+// aNat64PrefixOption
+
+void Nat64PrefixOption::Init(void)
+{
+    Clear();
+    SetType(kTypeNat64PrefixInfo);
+    SetSize(sizeof(Nat64PrefixOption));
+}
+
+const uint8_t Nat64PrefixOption::kPrefixLengths[] = {96, 64, 56, 48, 40, 32};
+
+void Nat64PrefixOption::SetLifetime(uint32_t aLifetime)
+{
+    uint32_t scaledLifetime = DivideAndRoundUp(aLifetime, kLifetimeScalingUnit);
+
+    mPrefixAttr = BigEndian::HostSwap16((BigEndian::HostSwap16(mPrefixAttr) & kPrefixLengthCodeMask) |
+                                        ClampToUint16(scaledLifetime << kScaledLifetimeOffset));
+}
+
+void Nat64PrefixOption::SetPrefixLengthCode(const uint8_t aPrefixLengthCode)
+{
+    mPrefixAttr = BigEndian::HostSwap16((BigEndian::HostSwap16(mPrefixAttr) & ~kPrefixLengthCodeMask) |
+                                        (aPrefixLengthCode & kPrefixLengthCodeMask));
+}
+
+Error Nat64PrefixOption::SetPrefix(const Prefix &aPrefix)
+{
+    Error error = kErrorInvalidArgs;
+
+    memcpy(mPrefixMsb, aPrefix.GetBytes(), sizeof(mPrefixMsb));
+
+    for (uint8_t code = 0; code < ClampToUint8(GetArrayLength(kPrefixLengths)); code++)
+    {
+        if (kPrefixLengths[code] == aPrefix.mLength)
+        {
+            SetPrefixLengthCode(code);
+            error = kErrorNone;
+            break;
+        }
+    }
+
+    return error;
+}
+
+Error Nat64PrefixOption::GetPrefix(Prefix &aPrefix) const
+{
+    Error   error            = kErrorNone;
+    uint8_t prefixLengthCode = GetPrefixLengthCode();
+
+    VerifyOrExit(prefixLengthCode < GetArrayLength(kPrefixLengths), error = kErrorParse);
+
+    aPrefix.Set(mPrefixMsb, kPrefixLengths[prefixLengthCode]);
+
+exit:
+    return error;
+}
+
+bool Nat64PrefixOption::IsValid(void) const
+{
+    return (GetLength() >= 2) && (GetPrefixLengthCode() < GetArrayLength(kPrefixLengths));
```

Comment:
Shall we check **GetLength() == 2**?
From RFC 8781: The sender MUST set the length to 2. The receiver MUST ignore the PREF64 option if the Length field value is not 2.

Reply:
Thanks for the explanation, @abtink!

Reply:
@abtink In this specific case, any backwards-compatibility with a future length increase is deliberately blocked by the RFC 8781 text. That implies that a future update of the Option is not foreseen and also not possible anymore. I guess that's because they will add a new number if such update would be needed and send the legacy option to serve legacy hosts. There's sufficient number space available for doing this strategy.

So I would just make the strict length check in line with the spec. Thread TLVs are different, because we require by spec that the length is extendible (and extra bytes ignored). And Thread group can't change this IETF decision anymore.